src/Microsoft.AspNetCore.Http/Internal/ParsingHelpers.cs

```diff
@@ -411,12 +411,11 @@ private static string DeQuote(string value)
                 throw new ArgumentNullException(nameof(headers));
             }
 
-            const NumberStyles styles = NumberStyles.AllowLeadingWhite | NumberStyles.AllowTrailingWhite;
             long value;
             var rawValue = headers[HeaderNames.ContentLength];
             if (rawValue.Count == 1 &&
                 !string.IsNullOrWhiteSpace(rawValue[0]) &&
-                long.TryParse(rawValue[0], styles, CultureInfo.InvariantCulture, out value))
+                HeaderUtilities.TryParseInt64(new StringSegment(rawValue[0]).Trim(), out value))
```

Comment:
We should do SetContentLength while we're at it, but I don't know if we have any optimized code for this yet. Kestrel has one but it's for hex.